### PR TITLE
:sparkles: Unban command

### PR DESCRIPTION
**Pull request by Gilbert142**
**Please describe the changes this PR makes and why it should be merged:**

**What does this PR do:**
- [ ] This PR changes internal functions, modules and/or utilities
  - [ ] This PR modifies the Extension API
- [x] This PR modifies commands
  - [ ] This PR changes command functions
  - [ ] This PR changes metadata for the command, updated in `commands.js`
  - [x] This PR removes and/or adds commands
- [ ] This PR modifies Web processing and/or content
  - [ ] This PR modifies static/ejs content
  - [ ] This PR modifies request processing (endpoint functions)
  - [ ] This PR modifies paths to existing resources
- [ ] This PR **only** includes non-code changes, like changes to default configurations, README, typos, etc.

### CODE REVIEW

File: Commands/Public/unban.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const { create: CreateModLog } = require("../../Modules/ModLog");
+const ArgParser = require("../../Modules/MessageUtils/Parser");
+
+const userFilter = query => ({ user }) => {
+	if (query.startsWith("<@") && query.endsWith(">")) {
+		query = query.slice(2, -1);
+	}
+
+	return user.id === query || user.tag === query || user.username === query;
```
Comment (vladfrangu) [inline Commands/Public/unban.js]: toLowerCase pls

File: Commands/Public/unban.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+module.exports = async ({ client, Constants: { Colors, Text }, configJS }, { serverDocument }, msg, commandData) => {
+	if (msg.suffix) {
+		const bans = await msg.guild.fetchBans();
+		let query, reason;
+		if (msg.suffix.includes("|")) [query, reason] = ArgParser.parseQuoteArgs(msg.suffix, "|");
```
Comment (vladfrangu) [inline Commands/Public/unban.js]: `let [query, ...reason] = ArgParser.parseQuoteArgs(msg.suffix, msg.suffix.includes('|') ? '|' : '')`

File: Commands/Public/unban.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		else query = msg.suffix;
+		if (!reason) reason = "Unspecified reason...";
+
+		if (!msg.guild.me.permissions.has("BAN_MEMBERS")) {
```
Comment (vladfrangu) [inline Commands/Public/unban.js:18]: Do this check before fetching

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by Gilbert142** [comment]
Approve or request more pls @vladfrangu

File: Commands/Public/unban.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+					color: Colors.SOFT_ERR,
+					description: "I couldn't find a matching user that's banned on this guild!",
+					footer: {
+						text: isNaN(query) || query.length !== 18 ? "Try unbanning by ID instead." : "",
+					},
+				},
+			});
+		}
+	} else {
+		msg.sendInvalidUsage(commandData, "Huh? Unban what?");
```
Comment (vladfrangu) [inline Commands/Public/unban.js]:
```suggestion
 		msg.sendInvalidUsage(commandData, "Huh? Unban who?");
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Pull request merged**